### PR TITLE
Clean-up guest builds

### PR DESCRIPTION
**Pull request by capossele**
As the `[target.'cfg(not(target_os = "zkvm"))'.dependencies]` does not seem to work for some reason, this PR exposes the feature `host`  (enabled by default) to prevent unnecessary deps ending up in the guest builds

### CODE REVIEW

**Review by Wollac — CHANGES_REQUESTED** [review]
Looking at the `Cargo.lock` it seems to work, but I do not understand the details:
- The TOML still contains `[target.'cfg(not(target_os = "zkvm"))'.dependencies]`. So this is still needed, but simply adding the optional will do the trick?
- Building the lib with `--no-default-features` now fails because the target OS is not `zkvm`, but all those dependencies are not included. This also causes problems with the rust-analyzer. Is there a way to fix this? One alternative I could think of is to remove all `#[cfg(not(target_os = "zkvm"))]` and manually enable/disable the `host` feature. However, I am not sure if this is a good idea.

File: lib/Cargo.toml [hunk 1]
```diff
@@ -15,15 +15,28 @@ thiserror = "1.0"
...
+host = [
+    "dep:chrono",
+    "dep:ethers-providers",
+    "dep:flate2",
+    "dep:log",
+    "dep:serde_json",
+    "dep:serde_with",
+    "dep:tokio",
+    "dep:zeth-primitives"
+]
```
Comment (Wollac) [inline lib/Cargo.toml:42]: `cargo sort` adds a trailing newline.

**Comment by capossele** [comment]
Yeah I'm not sure why `[target.'cfg(not(target_os = "zkvm"))'.dependencies]` does not really work as expected. It can be removed though if properly switching as you suggested from `#[cfg(not(target_os = "zkvm"))]` to #[cfg(feature = "host"))].
I've pushed a commit doing that.

Overall this is just a suggestion, maybe a bit hacky, so no harm in closing it if you guys are not convinced. Obviously would be better to understand why `[target.'cfg(not(target_os = "zkvm"))'.dependencies]` does not properly work

**Comment by capossele** [comment]
Actually I did not feel comfortable at removing the `#[cfg(not(target_os = "zkvm"))]`. So I ended up having them both

**Comment by Wollac** [comment]
It seems that `[target.'cfg(not(target_os = "zkvm"))'.dependencies]` actually works as expected, but the disabled dependencies show up in the `Cargo.lock` of the guest. However, they are resolved correctly. See: https://github.com/rust-lang/cargo/issues/10801

**Pull request closed without merging**